src/fheroes2/editor/editor_castle_details_window.cpp

```diff
@@ -576,6 +623,17 @@ namespace Editor
                     fheroes2::showStandardTextMessage( _( "Default Army" ), message, Dialog::ZERO );
                 }
             }
+            else if ( le.isMouseCursorPosInArea( buttonResetArmy.area() ) ) {
+                message = _( "Reset the army." );
+                if ( le.MouseClickLeft( buttonResetArmy.area() ) ) {
+                    castleArmy.Reset( false );
+                    armyBar.Redraw( display );
+                    display.render( dialogRoi );
```

Comment:
For this action we can render only the cleared army bar:
```suggestion
                    display.render( armyBar.GetArea() );
```